arch/arm/core/cortex_m/reset.S

```diff
@@ -57,6 +60,10 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__reset)
  */
 SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 
+#if defined(CONFIG_PLATFORM_SPECIFIC_INIT)
+    bl _PlatformInit
```

Comment:
Why do we need to call _PlatformInit this early?  Can't we just call it from nxp_lpc54114_init().  Also do we really need this implemented in ASM?

Reply:
The developer of RPMsg-Lite told me it needs to be run while the code executes from address 0x0000000. This comment may give some information:

`
/* Setup the reset handler pointer (PC) and stack pointer value.
             * This is used once the second core runs its startup code.
             * The second core first boots from flash (address 0x00000000)
             * and then detects its identity (Cortex-M0, slave) and checks
             * registers CPBOOT and CPSTACK and use them to continue the
             * boot process.
             * Make sure the startup code for current core (Cortex-M4) is
             * appropriate and shareable with the Cortex-M0 core!
             */ 
`

Reply:
This is not related to RPMsg-Lite, but to the architecture of LPC54114.
The secondary core in LPC54114 SOC needs to have a possibility of platform specific code in startup and cannot only use the generic cortex_m/reset.S

@galak Can you develop on why you disagree with this? Do you think other platforms may not need this?